### ISSUE TITLE
Loss of CustomXML parts/tags in XLSX & PPTX files

### ISSUE THREAD

**Issue by amaigaki** [ONLYOFFICE/DocumentServer#1564]
**Do you want to request a *feature* or report a *bug*?**
_Bug_

**What is the current behavior?**
When a PPTX or XLSX file having custom meta data is edited and saved back from DocumentServer, the customXml parts are lost in the said files. It works for DOCX files, my guess is because of ticket [792](https://github.com/ONLYOFFICE/DocumentServer/issues/792).
Files:
[Book.xlsx](https://drive.google.com/uc?export=download&id=1d7ZnE2uCqaqDZ8RHf_A6iaxBKd8XdvrZ)
[Presentation.pptx](https://drive.google.com/uc?export=download&id=124VQhLK3CjJjQHydd0qsWZKmzo9oZuSE)
[Document.docx](https://drive.google.com/uc?export=download&id=1js0w4fkA84uyoOsQ1JDPhwwz_SlFctqc)

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
1. Unzip PPTX or XLSX file having custom metadata (above given files) and observe the folder structure.
2. Open the file in DocumentServer, edit and save it back. 
3. Unzip the file and compare the folder structure with the previous one, the customXml folder is lost.

Before :
![image](https://user-images.githubusercontent.com/66057841/147068482-f80aefdf-da7f-4d23-9edf-cb1ee1a70142.png)

After edit and saveback :
![image](https://user-images.githubusercontent.com/66057841/147068596-cf2cb2a4-c787-4b44-9bd3-5030701b281d.png)

**What is the expected behavior?**
The custom meta data of the file should persist after editing and saveback in XLSX and PPTX formats also.

**Did this work in previous versions of DocumentServer?**
No

**DocumentServer version:**
6.3.1

**Operating System:**
Any

**Browser version:**
Any

**Comment by ShockwaveNN** [comment]
> DocumentServer version:
> 6.3.1

Hi, I think we fixed a lot of this issues at v6.4

See https://github.com/ONLYOFFICE/DocumentServer/issues/1443

So please update to latest stable v6.4.2 and see if this problem persists

**Comment by amaigaki** [comment]
Okay will check and let you know asap.

> > DocumentServer version:
> > 6.3.1
> 
> Hi, I think we fixed a lot of this issues at v6.4
> 
> See #1318
> 
> So please update to latest stable v6.4.2 and see if this problem persists

**Comment by amaigaki** [comment]
> > DocumentServer version:
> > 6.3.1
> 
> Hi, I think we fixed a lot of this issues at v6.4
> 
> See #1318
> 
> So please update to latest stable v6.4.2 and see if this problem persists

I checked, It is only fixed for DOCX files, not PPTX and XLSX files.

**Comment by ShockwaveNN** [comment]
Thanks for your issue, I've took a look any my findings are
1. For xlsx files seems that we already fixed loss of customXml. I've checked dev version v7.0.0 and seems it's not losing xlsx customXML. We hope we'll be ready to release it early next year
2. For pptx - your problem is reproduces on either stable version and v7.0.0. We indeed losing `customXML` in pptx. I've created issue 54673 in our private issue tracker

**Comment by amaigaki** [comment]
> Thanks for your issue, I've took a look any my findings are
> 
> 1. For xlsx files seems that we already fixed loss of customXml. I've checked dev version v7.0.0 and seems it's not losing xlsx customXML. We hope we'll be ready to release it early next year
> 2. For pptx - your problem is reproduces on either stable version and v7.0.0. We indeed losing `customXML` in pptx. I've created issue 54673 in our private issue tracker

Can I get the link of XLSX fix ?

**Comment by ShockwaveNN** [comment]
You can try using docker image `onlyoffice/4testing-documentserver-ee:7.0.0.96`

But please not that it's not stable version and can be contain any number of issues, but your xlsx downloaded fine

**Comment by amaigaki** [comment]
> You can try using docker image `onlyoffice/4testing-documentserver-ee:7.0.0.96`
> 
> But please not that it's not stable version and can be contain any number of issues, but your xlsx downloaded fine

Actually I was asking if I can get a link of the fixed changes so that we can port it into our my own version for the time being. Something like this 
[ONLYOFFICE/core@1a722a4](https://github.com/ONLYOFFICE/core/commit/1a722a46cbb8109eeb749ee64a1993736eec953d) 
[ONLYOFFICE/sdkjs@ecbe369](https://github.com/ONLYOFFICE/sdkjs/commit/ecbe369dfadf04a4a6b63de6383a9a46b7628987)

**Comment by ShockwaveNN** [comment]
Sorry, I don't think I remember when our dev team fixed it, so it took a lot of time to figure out
And I think you cannot just cherry-pick those commits in your version, because it may break other parts. I think it's better to wait for v7.0.0 release

**Comment by ShockwaveNN** [comment]
This issue is fixed for pptx files via
https://github.com/ONLYOFFICE/sdkjs/commit/55ec1abc35545ba91ac3ddfb41b020195f341aa6
https://github.com/ONLYOFFICE/core/commit/995b87e3bc495b09255eb6c26a672da3c7633033

And will be released as v7.2 in future (Sorry, no ETA)

**Comment by amaigaki** [comment]
> This issue is fixed for pptx files via [ONLYOFFICE/sdkjs@55ec1ab](https://github.com/ONLYOFFICE/sdkjs/commit/55ec1abc35545ba91ac3ddfb41b020195f341aa6) [ONLYOFFICE/core@995b87e](https://github.com/ONLYOFFICE/core/commit/995b87e3bc495b09255eb6c26a672da3c7633033)
> 
> And will be released as v7.2 in future (Sorry, no ETA)

Thanks for the update.

**Comment by teamoo** [comment]
This seems to be broken for docx files again on 7.1.1 @ShockwaveNN can you confirm that it is fixed in 7.2 for docx as well?

**Comment by ShockwaveNN** [comment]
@teamoo Yeah, thanks for noticifing, yes on v7.1.1 something is not right and item1.xml... files are present but empty

But on v7.2 they are ok
![image](https://user-images.githubusercontent.com/668524/188098887-89ab8578-1fea-4fca-8f40-0ddf063a084b.png)

**Comment by amaigaki** [comment]
> @teamoo Yeah, thanks for noticifing, yes on v7.1.1 something is not right and item1.xml... files are present but empty
> 
> But on v7.2 they are ok ![image](https://user-images.githubusercontent.com/668524/188098887-89ab8578-1fea-4fca-8f40-0ddf063a084b.png)

@ShockwaveNN, Thanks for the confirmation. has the release date been fixed for v7.2?

**Comment by ShockwaveNN** [comment]
@amaigaki We usually have no exact date for releases, but we planning to do it in September if everything went as planned

**Comment by cpot** [comment]
For important bugs/regressions it would be nice to have them fixed in minor release i.e. 7.1.2

**Comment by ShockwaveNN** [comment]
@cpot We are on final stage of preparing v7.2.0 so releasing v7.1.2 will postpone v7.2.0

But as I always say to you - you can always contact support@onlyoffice.com and they have more power to make some thing to happen

**Comment by ShockwaveNN** [comment]
DocumentServer v7.2 is released. This issue should be fixed

Feel free to comment or reopen it if you got further questions
